File: app/src/main/java/com/github/pockethub/android/ui/item/news/CreateEventItem.java
package com.github.pockethub.android.ui.item.news;

import android.support.annotation.NonNull;
import android.view.View;

import com.github.pockethub.android.ui.StyledText;
import com.github.pockethub.android.ui.view.OcticonTextView;
import com.github.pockethub.android.util.AvatarLoader;
import com.meisolsson.githubsdk.model.GitHubEvent;
import com.meisolsson.githubsdk.model.payload.CreatePayload;

public class CreateEventItem extends NewsItem {

    public CreateEventItem(AvatarLoader avatarLoader, GitHubEvent dataItem) {
        super(avatarLoader, dataItem);
    }

    @Override
    public void bind(@NonNull NewsItem.ViewHolder viewHolder, int position) {
        super.bind(viewHolder, position);
        viewHolder.icon.setText(OcticonTextView.ICON_CREATE);

        StyledText main = new StyledText();
        boldActor(main, getData());
        CreatePayload payload = (CreatePayload) getData().payload();

        main.append(" created ");
        String refType = payload.refType().name();
        main.append(refType);
        main.append(' ');

        if (!"Repository".equals(refType)) {
            main.append(payload.ref());
            main.append(" at ");
            boldRepo(main, getData());
        } else {
            boldRepoName(main, getData());
        }

        viewHolder.event.setText(main);
        viewHolder.details.setVisibility(View.GONE);
    }
}

File: app/src/main/java/com/github/pockethub/android/ui/item/news/CreateEventItem.kt
package com.github.pockethub.android.ui.item.news

import android.view.View
import androidx.core.text.buildSpannedString
import com.github.pockethub.android.ui.view.OcticonTextView
import com.github.pockethub.android.util.AvatarLoader
import com.meisolsson.githubsdk.model.GitHubEvent
import com.meisolsson.githubsdk.model.payload.CreatePayload
import com.xwray.groupie.kotlinandroidextensions.ViewHolder
import kotlinx.android.synthetic.main.news_item.*

class CreateEventItem(
        avatarLoader: AvatarLoader,
        gitHubEvent: GitHubEvent
) : NewsItem(avatarLoader, gitHubEvent) {

    override fun bind(holder: ViewHolder, position: Int) {
        super.bind(holder, position)
        holder.tv_event_icon.text = OcticonTextView.ICON_CREATE
        holder.tv_event.text = buildSpannedString {
            boldActor(this, gitHubEvent)
            val payload = gitHubEvent.payload() as CreatePayload?

            val refType: String? = payload?.refType()?.name?.toLowerCase()
            append(" created $refType ")

            if ("repository" != refType) {
                append("${payload?.ref()} at ")
                boldRepo(this, gitHubEvent)
            } else {
                boldRepoName(this, gitHubEvent)
            }
        }
        holder.tv_event_details.visibility = View.GONE
    }
}
